pybamm/models/submodels/current_collector/effective_resistance_current_collector.py

```diff
@@ -1,196 +1,401 @@
+
+        def V_cc_dim(t, z, y=None):
+            return U_ref + V_cc(t, z, y=y) * pot_scale
+
+        processed_vars = {
+            "Negative current collector potential": phi_s_cn,
+            "Negative current collector potential [V]": phi_s_cn_dim,
+            "Positive current collector potential": phi_s_cp,
+            "Positive current collector potential [V]": phi_s_cp_dim,
+            "Local current collector potential difference": V_cc,
+            "Local current collector potential difference [V]": V_cc_dim,
+            "Terminal voltage": V,
+            "Terminal voltage [V]": V_dim,
+        }
+        return processed_vars
+
+    @property
+    def default_parameter_values(self):
+        return pybamm.ParameterValues(chemistry=pybamm.parameter_sets.Marquis2019)
+
+    @property
+    def default_geometry(self):
+        return pybamm.Geometry("1D current collector")
+
+    @property
+    def default_var_pts(self):
+        var = pybamm.standard_spatial_vars
+        return {var.y: 32, var.z: 32}
+
+    @property
+    def default_submesh_types(self):
+        return {"current collector": pybamm.MeshGenerator(pybamm.Uniform1DSubMesh)}
+
+    @property
+    def default_spatial_methods(self):
+        return {"current collector": pybamm.FiniteVolume()}
+
+    @property
+    def default_solver(self):
+        return pybamm.CasadiAlgebraicSolver()
+
+
+class EffectiveResistance1D(EffectiveResistance):
+    """
+    A model which calculates the effective Ohmic resistance of the 1D current
     collectors in the limit of large electrical conductivity.
-    Note:  This submodel should be solved before a one-dimensional model to calculate
-    and return the effective current collector resistance.
+
+    **Extends:** :class:`pybamm.EffectiveResistance`
+    """
+
+    def __init__(
+        self, options=None, name="Effective resistance in current collector model (1D)"
+    ):
+        super().__init__(options, name)
+
+    def set_boundary_conditions(self, variables):
+        R_cn_scaled = variables["Scaled negative current collector resistance"]
+        R_cp_scaled = variables["Scaled positive current collector resistance"]
+        self.boundary_conditions = {
+            R_cn_scaled: {"negative tab": (0, "Dirichlet"), "no tab": (0, "Neumann")},
+            R_cp_scaled: {"positive tab": (0, "Dirichlet"), "no tab": (0, "Neumann")},
+        }
+
+    def average(self, variable):
+        "Compute the average over the 1D current collector"
+        return pybamm.z_average(variable)
+
+
+class EffectiveResistance2D(EffectiveResistance):
+    """
+    A model which calculates the effective Ohmic resistance of the 2D current
+    collectors in the limit of large electrical conductivity. Note that this
+    formulation assumes uniform potential across the tabs.
+    See :class:`pybamm.AlternativeEffectiveResistance2D` for the formulation that
+    assumes a uniform current density at the tabs.
```

Comment:
maybe emphasize `potential` and `current density` in this docstring

Reply:
good point

> I got a bit confused by what EffectiveResistance did (for a while I thought it was a 0D model). I think it would be easier to understand if that model took an option dimensionality which specifies 1D behaviour vs 2D behaviour, instead of subclassing. This is more consistent with how we have been reformatting the submodels

Agreed, will change to this. Also, in this case these models will then work with simulation, so maybe best to leave base model alone